cpp/include/raft/comms/detail/mpi_comms.hpp

```diff
@@ -119,8 +126,11 @@ class mpi_comms : public comms_iface {
 
     // initializing NCCL
     RAFT_NCCL_TRY(ncclCommInitRank(&nccl_comm_, size_, id, rank_));
+    initialize();
   }
 
+  void initialize() { sendbuff_ = status_.data(); }
```

Comment:
May better set status_ to 0 to avoid status_ holding garbage. Unlikely to affect any program behavior, but I guess this is a safer practice (especially considering that performance overhead is negligible).

And if we use rmm::device_scalar instead of rmm::device_uvector, we can simply call set_value_to_zero_async (https://github.com/rapidsai/rmm/blob/branch-22.04/include/rmm/device_scalar.hpp#L215).